File: src/main/java/me/sul/crackshotaddition/events/WeaponProjectileTrailEvent.java
package me.sul.crackshotaddition.events;

import lombok.Getter;
import lombok.Setter;
import org.bukkit.Particle;
import org.bukkit.entity.Entity;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;

@Getter
public class WeaponProjectileTrailEvent extends Event {
	private static final HandlerList handlers = new HandlerList();
	private final Entity entity;
	private @Setter Particle particle;
	private final String weaponTitle;
	
	public WeaponProjectileTrailEvent(Entity entity, String weaponTitle, Particle particle) {
		this.particle = particle;
		this.weaponTitle = weaponTitle;
		this.entity = entity;
	}

	@Override
	public HandlerList getHandlers() {
		return handlers;
	}
	public static HandlerList getHandlerList()
	{
		return handlers;
	}

}

File: src/main/kotlin/kr/sul/crackshotaddition/events/WeaponProjectileTrailEvent.kt
package kr.sul.crackshotaddition.events

import org.bukkit.Particle
import org.bukkit.entity.Entity
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

// Scav때문에 player이 아닌 entity로 해 놓은 듯
class WeaponProjectileTrailEvent(val entity: Entity, val parentNode: String, var particle: Particle) : Event() {

    override fun getHandlers(): HandlerList {
        return handlerList
    }
    companion object {
        @JvmStatic
        val handlerList = HandlerList()
    }
}